src/main/java/cubicchunks/world/cube/Cube.java

```diff
@@ -417,7 +419,7 @@ public void tickCubeServer(BooleanSupplier tryToTickFaster, Random rand) {
     public void randomTick(WorldServer worldServer, Random rand) {
         this.updateLCG = this.updateLCG * 3 + 1013904223;
         int j1 = updateLCG >> 2;
-        int localX = j1 & 15;
+        int localX = j1 & 15;           // TODO: Why not use AddressTools.getLocalX() instead? Uses the first 12 bits, instead of the first 4, 9 through 12 and 17 through 20.
         int localY = j1 >> 8 & 15;
         int localZ = j1 >> 16 & 15;
```

Comment:
```int localZ = blockToLocal(j1 >> 16)``` will work fine too. Unless you will decide to make Cube.size = 1024 or something.

Reply:
AddressTools probably can be used just fine here. The exact order of whoch bits are used for what doesn't really matter.